index.html

```diff
@@ -500,6 +509,9 @@ <h3>The <dfn>observe()</dfn> method</h3>
               [=Queue a global task=] on the [=PressureObserver task source=] given |document|'s
               [=relevant global object=] |relevantGlobal| to run these steps:
               <ol>
+                <li>
+                  If |promise| was rejected, abort these steps.
```

Comment:
Isn't there a short window between "If |source:PressureSource| is not a [=supported source type=] ..." and "Activate [=data delivery=]" where disconnect or unobserve could run leaving you in a state with no promises but activated data delivery?

Reply:
You might be right.

Rejecting the promise doesn't stop the parallel steps, so we should add a promise test as well before the activation right?

Reply:
But as it is running in parallel, I probably don't have access to the promise to check its state

Reply:
As we activate "in parallel", I think we may in some care cases activate the platform collector even without having any observers. It is a bit separate from the wrong state this PR is solving, and I think the easiest way to solve it would be to check if there are active observers when we deliver readings and if not, deactivate it.

Reply:
We can also just deactivate it immediately when we do the main thread steps.